### ISSUE TITLE
Zero shot for Prompt.

### ISSUE THREAD

**Issue by GuodongFan** [adf1178/PT4Code#1]
How is the vanilla soft prompt of zero-shot data obtained in the experiment? Is the hard prompt used directly?

**Comment by GuodongFan** [comment]
And can you give an example of what is a hard prompt template for T5?

**Comment by adf1178** [comment]
Q1: We don't experiment with the vanilla soft prompt in zero-shot because there is no semantic in the template, thus the model can not understand it. We only try hard templates.
Q2: Hard prompt for CodeT5 is similar with the muti-task training of CodeT5 such as "Summarize Python: <CODE>"

**Comment by GuodongFan** [comment]
Is the defect missing the preprocess file?

**Comment by adf1178** [comment]
Preproceccing Defect detection is the same as CodeXGlue

**Comment by GuodongFan** [comment]
For 4.3.3 what is the prefix of soft prompts refer to ? is [soft]*n [X][Z] ? why it is needed for 100. In my opinion, for the hard prompt, the prefix is only a few words, for example, Generate comment for ...

**Comment by GuodongFan** [comment]
In the paper, cross-domain scenario, is the experiments fine-tune on Python and Java dataset and then further use small data to fine-tune to prompt tuning on the target dataset? or when fine-tune/prompt tuning on Python and Java, and then zero shot testing on target dataset?

**Comment by adf1178** [comment]
Fine-tuning on Python and Java, then zero shot testing on target test set
